core/data_category.js

```diff
@@ -374,6 +401,61 @@ Blockly.DataCategory.addHideList = function(xmlList, variable) {
   Blockly.DataCategory.addBlock(xmlList, variable, 'data_hidelist', 'LIST');
 };
 
+/**
+ * Construct and add a data_dictcontents block to xmlList.
+ * @param {!Array.<!Element>} xmlList Array of XML block elements.
+ * @param {?Blockly.VariableModel} variable Variable to select in the field.
+ */
+Blockly.DataCategory.addDataDict = function(xmlList, variable) {
+  // <block id="variableId" type="data_dictcontents">
+  //    <field name="DICT">variablename</field>
+  // </block>
+  Blockly.DataCategory.addBlock(xmlList, variable, 'data_dictcontents', 'DICT');
+  // In the flyout, this ID must match variable ID for monitor syncing reasons
+  xmlList[xmlList.length - 1].setAttribute('id', variable.getId());
+};
+
+/**
+ * Construct and add a data_addtodict block to xmlList.
+ * @param {!Array.<!Element>} xmlList Array of XML block elements.
+ * @param {?Blockly.VariableModel} variable Variable to select in the field.
+ */
+Blockly.DataCategory.addAddToDict = function(xmlList, variable) {
+  // <block type="data_addtodict">
+  //   <field name="DICT" variabletype="dict" id="">variablename</field>
+  //   <value name="ITEM">
+  //     <shadow type="text">
+  //       <field name="TEXT">thing</field>
+  //     </shadow>
+  //   </value>
+  //   <value name="KEY">
+  //     <shadow type="text">
+  //       <field name="TEXT">key</field>
+  //     </shadow>
+  //   </value>
+  // </block>
+  Blockly.DataCategory.addBlock(xmlList, variable, 'data_addtodict', 'DICT',
+      ['ITEM', 'text', Blockly.Msg.DEFAULT_DICT_ITEM], ['KEY', 'text', Blockly.Msg.DEFAULT_DICT_KEY]);
```

Comment:
This does not work as I expected. The second array does not control the second parameter of the block, which is the key. Both the value and the key are "thing" by default, which corresponds to `Blockly.Msg.DEFAULT_DICT_ITEM`.

Either find out how that actually works or live with it and remove the second parameter (if it actually works without it).

Reply:
Caused by a bug in the existing codebase. Fixed by 
e649c88. Sent a PR to upstream.